busybox image: add symlink for /bin/mkfifo

justinsb opened the pull request:
This makes the bazel image more similar to the bash build, which uses
the image from the docker registry.

mkfifo is used by e.g. kops to send logs to both a /var/log file and to
stdout (which is then captured by docker).

```release-note
Add /bin/mkfifo symlink to bazel build for busybox, so that CI builds have /bin/tee
```

justinsb committed:
busybox image: add symlink for /bin/mkfifo

This makes the bazel image more similar to the bash build, which uses
the image from the docker registry.

mkfifo is used by e.g. kops to send logs to both a /var/log file and to
stdout (which is then captured by docker).

justinsb commented:
This addresses #59263.  Note this is only against the 1.7 branch, and I believe this image is only used by CI (not by our releases).  In 1.8 we switched how the image was built, so it doesn't need to be applied to 1.8+ (in fact, it can't be IIRC)

wojtek-t commented:
/lgtm
/approve no-issue

justinsb commented:
/retest

wojtek-t commented:
@justinsb - it seems it didn't fix the problem - kops presubmit is still failing with this PR.

justinsb committed:
Force creation of /tmp directory in busybox image

We create a symlink to hostname, to force creation of the /tmp
directory.  There doesn't seem to be a direct way of creating a
directory in the docker_build task in bazel.

justinsb commented:
Had to add a symlink under /tmp, to force creation of the `/tmp` directory (in which we are creating that FIFO pipe).  I don't love it, but it seems the least-bad option that I came up with.  This should only be used for the bazel CI image for 1.7, as the release version uses a "real" docker image and we pull that base image from 1.8 for bazel.

wojtek-t commented:
/lgtm
